openwisp_controller/config/admin.py

```diff
@@ -533,6 +534,15 @@ def get_inlines(self, request, obj):
                 inlines.append(inline)
         return inlines
 
+    def reversion_register(self, model, **options):
+        if model == Device:
+            options['follow'] = (
+                *(options['follow']),
+                'deviceconnection_set',
+                'devicelocation',
+            )
```

Comment:
Will it be possible to add support for other relations in openwisp-firmware-upgrader and openwisp-monitoring, right?
We'll have to do the same with subnet division rule.